spec/datadog/tracing/contrib/utils/quantization/http_spec.rb

```diff
@@ -271,6 +278,101 @@
+        end
+
+        context 'with a matching, URL-encoded JSON substring' do
+          let(:query) { 'json=%7B%20%22sign%22%3A%20%22%7B0x03cb9f67%2C0xdbbc%2C0x4cb8%2C%7B0xb9%2C0x66%2C0x32%2C0x99%2C0x51%2C0xe1%2C0x09%2C0x34%7D%7D%22%7D' }
+          let(:options) { { obfuscate: :internal } }
+
+          it { is_expected.to eq('json=%7B%20%22<redacted>%7D') }
+        end
+
+        context 'with a reduced show option overlapping with a potential obfuscation match' do
+          let(:query) { 'pass=03cb9f67-dbbc-4cb8-b966-329951e10934&key2=val2&key3=val3' }
+          let(:options) { { show: ['pass', 'key2'], obfuscate: :internal } }
+
+          it { is_expected.to eq('<redacted>&key2=val2&key3') }
+        end
+
+        context 'with a reduced show option distinct from a potentail obfuscation match' do
+          let(:query) { 'pass=03cb9f67-dbbc-4cb8-b966-329951e10934&key2=val2&key3=val3' }
+          let(:options) { { show: ['key2'], obfuscate: :internal } }
+
+          it { is_expected.to eq('pass&key2=val2&key3') }
+        end
+
+        context 'with an exclude option overlapping with a potential obfuscation match' do
+          let(:query) { 'pass=03cb9f67-dbbc-4cb8-b966-329951e10934&key2=val2&key3=val3' }
+          let(:options) { { exclude: ['key2'], obfuscate: :internal } }
+
+          it { is_expected.to eq('<redacted>&key3=val3') }
+        end
+
+        context 'with an exclude option distinct from a potential obfuscation match' do
+          let(:query) { 'pass=03cb9f67-dbbc-4cb8-b966-329951e10934&key2=val2&key3=val3' }
+          let(:options) { { exclude: ['pass'], obfuscate: :internal } }
+
+          it { is_expected.to eq('key2=val2&key3=val3') }
+        end
+      end
+
+      context 'and an obfuscate with custom options' do
+        context 'with regex: :internal' do
+          let(:query) { 'pass=03cb9f67-dbbc-4cb8-b966-329951e10934&key2=val2&key3=val3' }
+          let(:options) { { obfuscate: { regex: :internal } } }
+
+          it { is_expected.to eq('<redacted>&key2=val2&key3=val3') }
+        end
+
+        context 'with a custom regex' do
+          let(:query) { 'pass=03cb9f67-dbbc-4cb8-b966-329951e10934&key2=val2&key3=val3' }
+          let(:options) { { obfuscate: { regex: /key2=[^&]+/ } } }
+
+          it { is_expected.to eq('pass=03cb9f67-dbbc-4cb8-b966-329951e10934&<redacted>&key3=val3') }
+        end
+
+        context 'with a custom replacement' do
+          let(:query) { 'pass=03cb9f67-dbbc-4cb8-b966-329951e10934&key2=val2&key3=val3' }
+          let(:options) { { obfuscate: { with: 'NOPE' } } }
+
+          it { is_expected.to eq('NOPE&key2=val2&key3=val3') }
+        end
+
+        context 'with an empty replacement' do
+          let(:query) { 'pass=03cb9f67-dbbc-4cb8-b966-329951e10934&key2=val2&key3=val3' }
+          let(:options) { { obfuscate: { with: '' } } }
+
+          it { is_expected.to eq('&key2=val2&key3=val3') }
```

Comment:
This is kind of awkward, I don't think it does what a user would expect.

Reply:
Seems fine to me, the `pass=...` was removed. What's your concern here? The leading `&`?

Reply:
Yup, not really a "concern" since it works, but the uncleaniness tickles my OCD ;)